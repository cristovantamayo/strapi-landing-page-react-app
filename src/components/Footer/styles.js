import styled, { css } from "styled-components";
import { theme } from "../../styles/theme";

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    a {
      color: inherit;
      text-decoration: nome;
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
