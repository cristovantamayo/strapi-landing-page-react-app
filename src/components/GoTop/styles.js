import styled, { css } from "styled-components";

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4.8rem;
    height: 4.8rem;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;
    opacity: 0.7;
    transition: all 0.3s ease-in-out;

    &:hover {
      opacity: 1;
      background: red;
    }
  `}
`;
