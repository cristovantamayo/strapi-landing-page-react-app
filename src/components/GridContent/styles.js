import styled, { css } from "styled-components";

export const Container = styled.div`
  ${({ theme }) => css`
    text-align: center;
    max-width: 60rem;
    margin: 0 auto;

    @media ${theme.media.lessMedium} {
      max-width: 40rem;
    }
  `}
`;

export const Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.xHuge};
  `}
`;
