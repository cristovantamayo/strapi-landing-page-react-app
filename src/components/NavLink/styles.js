import styled, { css } from "styled-components";

export const Container = styled.div`
  ${({ theme }) => css`
    background: #fff;
    display: flex;
    flex-flow: row wrap;

    @media ${theme.media.lessMedium} {
      flex-flow: column wrap;
      align-content: center;
    }
  `}
`;
