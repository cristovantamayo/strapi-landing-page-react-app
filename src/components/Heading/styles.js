import styled, { css } from "styled-components";
import { theme } from "../../styles/theme";

const titleSizes = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.font.sizes.large};
  `,
  big: (theme) => css`
    font-size: ${theme.font.sizes.xLarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.font.sizes.huge};
    ${mediaFont(theme)};
  `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lessMedium} {
    font-size: ${theme.font.sizes.xLarge};
  }
`;

const titleCase = (upperCase) => css`
  text-transform: ${upperCase ? "none" : "uppercase"};
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, upperCase }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSizes[size](theme)}
    ${titleCase(upperCase)}
  `}
`;
