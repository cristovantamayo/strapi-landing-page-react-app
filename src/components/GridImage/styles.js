import styled, { css } from "styled-components";
import { Title as HeadingContainer } from "../Heading/styles";
import { Container as TextComponent } from "../TextComponent/styles";

export const Container = styled.div`
  ${({ theme }) => css`
    max-width: 128rem;
    margin: 0 auto;
    left: 0;
    right: 0;
    & > ${TextComponent} {
      margin-bottom: ${theme.spacings.xHuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    max-width: 120rem;
    margin: 0 auto;

    @media ${theme.media.lessMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
    width: 100%;
    height: 280px;
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    height: 100%;
    width: 100%;
    object-fit: cover;

    &:hover {
      transform: scale(1.2) rotate(10deg);
      transition: transform 0.5s;
    }
  `}
`;
