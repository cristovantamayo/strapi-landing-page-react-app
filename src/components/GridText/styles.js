import styled, { css } from "styled-components";
import { Title as HeadingContainer } from "../Heading/styles";
import { Container as TextComponent } from "../TextComponent/styles";

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;
    max-width: 120rem;
    ${TextComponent} {
      margin-bottom: ${theme.spacings.xHuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.xHuge};
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      position: relative;
      left: 5rem;
    }

    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.font.sizes.xHuge};
      top: -3.5rem;
      left: -4rem;
      transform: rotate(10deg);
  `}
`;
