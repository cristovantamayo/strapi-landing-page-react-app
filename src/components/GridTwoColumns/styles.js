import styled, { css } from "styled-components";
import { Title } from "../Heading/styles";

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    justify-content: center;
    gap: ${theme.spacings.large};

    @media ${theme.media.lessMedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }

    ${Title} {
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.media.lessMedium} {
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`
    max-width: 80rem;
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css``}
`;
