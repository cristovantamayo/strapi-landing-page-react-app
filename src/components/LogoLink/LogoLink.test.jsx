import { screen } from "@testing-library/react";
import { describe, expect } from "@jest/globals";
import { renderTheme } from "../../styles/render-theme";
import { LogoLink } from ".";

describe("<LogoLink />", () => {
  it("should render text logo", () => {
    renderTheme(
      <LogoLink link="#target" text="Olá Mundo">
        Children
      </LogoLink>,
    );
    const heading = screen.getByRole("heading", { name: "Olá Mundo" });
    expect(heading.firstChild).toHaveAttribute("href", "#target");
    expect(
      screen.getByRole("heading", { name: "Olá Mundo" }),
    ).toBeInTheDocument();
    // a another way to
    expect(screen.getByRole("link", { name: "Olá Mundo" })).toHaveAttribute(
      "href",
      "#target",
    );
  });

  it("should render image logo", () => {
    renderTheme(
      <LogoLink link="#target" text="Olá Mundo" srcImg="image.jpg">
        Children
      </LogoLink>,
    );

    expect(screen.getByRole("img", { name: "Olá Mundo" })).toHaveAttribute(
      "src",
      "image.jpg",
    );
    // a another way to
    expect(screen.getByAltText("Olá Mundo")).toHaveAttribute(
      "src",
      "image.jpg",
    );
  });

  it("should match snapshot", () => {
    const { container } = renderTheme(
      <LogoLink link="#target" text="Olá Mundo" srcImg="image.jpg">
        Children
      </LogoLink>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
