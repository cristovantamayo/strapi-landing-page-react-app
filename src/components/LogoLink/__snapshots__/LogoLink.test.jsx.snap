// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoLink /> should match snapshot 1`] = `
.c1 {
  display: flex;
  align-items: center;
  text-decoration: none;
  color: inherit;
}

.c1 >img {
  height: 6rem;
}

.c0 {
  color: #0A1128;
  font-size: 2.4rem;
  text-transform: none;
}

<h1
  class="c0"
>
  <a
    class="c1"
    href="#target"
  >
    <img
      alt="OlÃ¡ Mundo"
      src="image.jpg"
    />
  </a>
</h1>
`;
